proxy/http/HttpSM.cc

```diff
@@ -502,25 +502,28 @@ HttpSM::attach_client_session(ProxyTransaction *client_vc, IOBufferReader *buffe
 
   // Collect log & stats information. We've already verified that the netvc is !nullptr above,
   // and netvc == ua_txn->get_netvc().
-  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
 
   is_internal       = netvc->get_is_internal_request();
   mptcp_state       = netvc->get_mptcp_state();
   client_tcp_reused = !(ua_txn->is_first_transaction());
 
-  if (ssl_vc != nullptr) {
-    client_connection_is_ssl = true;
-    client_ssl_reused        = ssl_vc->getSSLSessionCacheHit();
-    const char *protocol     = ssl_vc->getSSLProtocol();
-    client_sec_protocol      = protocol ? protocol : "-";
-    const char *cipher       = ssl_vc->getSSLCipherSuite();
-    client_cipher_suite      = cipher ? cipher : "-";
-    const char *curve        = ssl_vc->getSSLCurve();
-    client_curve             = curve ? curve : "-";
-    if (!client_tcp_reused) {
-      // Copy along the TLS handshake timings
-      milestones[TS_MILESTONE_TLS_HANDSHAKE_START] = ssl_vc->sslHandshakeBeginTime;
-      milestones[TS_MILESTONE_TLS_HANDSHAKE_END]   = ssl_vc->sslHandshakeEndTime;
+  SSLNetVConnection *ssl_vc = nullptr;
+  if (ua_txn->is_tls()) {
```

Comment:
I assume this is to avoid unnecessary dynamic_dast. Why don't you add `is_tls` or something to NetVC?

Reply:
I had always thought dynamic_cast was done inline and was fast.  But it actually requires a call into the runtime support library:  https://godbolt.org/z/u2u-ax .  I think @bryancall saw that dynamic_cast was slow in some performance testing.

Reply:
That might make sense.  I think we have some issues of lack of certainty about object lifetimes.  I think, at least when you reach the point of creating log entries, the NetVC object no longer exists.  The problem is that, for logging and maybe other purposes, the session, txn, etc objects correspond to sessions, txns, etc. that existed in the past, not currently.